### PR TITLE
Experiment with scripts to manage GH labels

### PR DESCRIPTION
**Pull request by tresat**
- create_labelowners.groovy will process the labels.csv file created from downloading the Google sheet as a CSV file and create a LABELOWNERS file to live along side CODEOWNERS, in a similar format.
- verify_labels.sh will fetch all unknown labels not referenced in labels.csv file.

Still requires manually downloading the labels Google sheet into the .csv file.

### CODE REVIEW

File: .github/LABELOWNERS [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+in:dependency-locking                        @gradle/bt-jvm
+in:dependency-management                     @gradle/bt-unassigned-maintainers
+in:dependency-remoting                       @gradle/bt-jvm
+in:dependency-resolution                     @gradle/bt-jvm
+in:dependency-verification                   @gradle/bt-jvm
+in:dependency-version-catalog                @gradle/bt-jvm
+in:deprecation-logging                       @gradle/bt-core-runtime-maintainers
+in:distribution-plugin                       @gradle/bt-jvm
+in:documentation                             @gradle/bt-unassigned-maintainers
+in:documentation-infrastructure              @gradle/null
```
Comment (big-guy) [inline .github/LABELOWNERS:56]: `@gradle/null`?

File: .github/LABELOWNERS [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+in:input-normalization                       @gradle/bt-execution
+in:integrations                              @gradle/bt-unassigned-maintainers
+in:internal-caches                           @gradle/bt-unassigned-maintainers
+in:invoking-gradle                           @gradle/bt-unassigned-maintainers
+in:jacoco-plugin                             @gradle/bt-jvm
+in:java-plugins                              @gradle/bt-jvm
+in:javadoc-task                              @gradle/bt-jvm
+in:jvm-ecosystem                             @gradle/bt-jvm
+in:kotlin-dsl                                @gradle/bt-configuration-cache
+in:learning                                  @gradle/null
```
Comment (ov7a) [inline .github/LABELOWNERS:79]: 🤔 there is no such label?

File: .github/LABELOWNERS [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+in:file-tasks                                @gradle/bt-execution
+in:gradle-module-metadata                    @gradle/bt-jvm
+in:groovy-dsl                                @gradle/bt-configuration-cache
+in:groovy-plugin                             @gradle/bt-jvm
+in:ide                                       @gradle/bt-ide-experience
+in:idea-plugin                               @gradle/bt-ide-experience
+in:init-scripts                              @gradle/bt-configuration-cache
+in:input-normalization                       @gradle/bt-execution
+in:integrations                              @gradle/bt-unassigned-maintainers
+in:internal-caches                           @gradle/bt-unassigned-maintainers
```
Comment (ov7a) [inline .github/LABELOWNERS:72]: 🤔 These two labels don't exist.
Is the label list up-to-date?

File: .github/issue_labels/create_labelowners.groovy [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+def codeOwnersList = []
+
+def platformsMap = [
+    'No platform': 'bt-unassigned-maintainers',
+    'JVM': 'bt-jvm',
+    'Core/execution': 'bt-execution',
+    'Core/runtime': 'bt-core-runtime-maintainers',
+    'Mixed': 'bt-unassigned-maintainers',
+    'GE integration': 'bt-build-scan',
```
Comment (ov7a) [inline .github/issue_labels/create_labelowners.groovy:36]: 💅 DV integration

File: .github/issue_labels/create_labelowners.groovy [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    'Core/configuration': 'bt-configuration-cache',
+    'Build infrastructure': 'bt-developer-productivity',
+    'No platform': 'bt-unassigned-maintainers',
+    '': 'bt-unassigned-maintainers',
```
Comment (ov7a) [inline .github/issue_labels/create_labelowners.groovy:40]: ❓ Why do we need multiple special values when we can fix them in the CSV?

File: .github/issue_labels/labels.csv [hunk 6]
```diff
@@ -0,0 +1,112 @@
+Label name,Issues,Description,SME,Team,Platform,Blanks,Issue count
```
Comment (ov7a) [inline .github/issue_labels/labels.csv:1]: 🤔 I assume this is taken from the Google doc, verbatim.
I think we should have a single source of truth, and it should be located in the repo, making the doc obsolete.

File: .github/issue_labels/unknown_labels.txt [hunk 7]
```diff
@@ -0,0 +1,127 @@
+1 min review
```
Comment (ov7a) [inline .github/issue_labels/unknown_labels.txt:1]: 🤔 what is the purpose of this file? I don't think we want to update it every time new label is introduced

File: .github/issue_labels/unknown_labels.txt [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+good first issue
+has:reproducer
+has:workaround
+is:blocked
+is:blocker
+p:lazy-migration
+pending:dco
+pending:pr-feedback
+pending:reproducer
+platform:bt-ge-build-cache
```
Comment (ov7a) [inline .github/issue_labels/unknown_labels.txt:104]: 🤔 probably it's a good time to revisit this platform:* labels
IIUC, nobody is using them.

However, we can use them as an aggregators for in: labels and assign platforms/owners

File: .github/issue_labels/verify_labels.groovy [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+// Read previously unknown labels
+def existingUnknownLabels = outputFile.readLines()
+
+def unknownLabels = [] as Set
+unknownLabels.addAll(newLabels*.name)
+unknownLabels.addAll(existingUnknownLabels)
+unknownLabels.removeAll(knownLabels)
+
+// Write the results
+outputFile.withWriter { writer ->
```
Comment (ov7a) [inline .github/issue_labels/verify_labels.groovy:33]: ❌ if I read this correctly, the outputFile is overwritten every time, meaning data loss for previous label pages.

**Pull request closed without merging**